diff --color -Naur pacman-5.2.1.orig/meson.build pacman-5.2.1/meson.build
--- pacman-5.2.1.orig/meson.build	2019-11-01 01:56:41.000000000 +0100
+++ pacman-5.2.1/meson.build	2020-05-12 23:50:53.732238004 +0200
@@ -47,7 +47,7 @@
 
 bashcompletion = dependency('bash-completion', required : false)
 if bashcompletion.found()
-  BASHCOMPDIR = bashcompletion.get_pkgconfig_variable('completionsdir')
+  BASHCOMPDIR = bashcompletion.get_pkgconfig_variable('completionsdir', define_variable: ['datadir', DATAROOTDIR])
 else
   BASHCOMPDIR = join_paths(DATAROOTDIR, 'bash-completion/completions')
 endif
@@ -322,7 +322,8 @@
   # https://github.com/mesonbuild/meson/issues/3937
   objects : libcommon.extract_all_objects(),
   include_directories : includes,
-  dependencies : alpm_deps)
+  dependencies : alpm_deps,
+  install : false)
 
 libalpm = library(
   'alpm',
@@ -331,17 +332,17 @@
   dependencies : alpm_deps,
   install : true)
 
-install_headers(
-  'lib/libalpm/alpm.h',
-  'lib/libalpm/alpm_list.h')
-
-pkgconfig = import('pkgconfig')
-pkgconfig.generate(
-  libalpm,
-  name : 'libalpm',
-  description : 'Arch Linux package management library',
-  version : libalpm_version,
-  url : 'http://www.archlinux.org/pacman/')
+#install_headers(
+#  'lib/libalpm/alpm.h',
+#  'lib/libalpm/alpm_list.h')
+
+#pkgconfig = import('pkgconfig')
+#pkgconfig.generate(
+#  libalpm,
+#  name : 'libalpm',
+#  description : 'Arch Linux package management library',
+#  version : libalpm_version,
+#  url : 'http://www.archlinux.org/pacman/')
 
 pacman_bin = executable(
   'pacman',
diff --color -Naur pacman-5.2.1.orig/scripts/meson.build pacman-5.2.1/scripts/meson.build
--- pacman-5.2.1.orig/scripts/meson.build	2019-08-12 03:26:42.000000000 +0200
+++ pacman-5.2.1/scripts/meson.build	2020-05-12 23:46:11.035405868 +0200
@@ -78,11 +78,11 @@
 
 subdir('libmakepkg')
 
-configure_file(
-	configuration : { 'libmakepkgdir': LIBMAKEPKGDIR, 'PACKAGE_VERSION': PACKAGE_VERSION },
-	input : 'libmakepkg.pc.in',
-	output : '@BASENAME@',
-	install_dir : join_paths(DATAROOTDIR, 'pkgconfig'))
+#configure_file(
+#	configuration : { 'libmakepkgdir': LIBMAKEPKGDIR, 'PACKAGE_VERSION': PACKAGE_VERSION },
+#	input : 'libmakepkg.pc.in',
+#	output : '@BASENAME@',
+#	install_dir : join_paths(DATAROOTDIR, 'pkgconfig'))
 
 custom_target(
   'bash_completion',
